tutorials/graph-theory/binary-search-tree.md

```diff
@@ -1,12 +1,391 @@
 ---
-title: 'Binary Search Tree'
-description: 'Coming Soon!'
+title: "Binary Search Tree (BST)"
+description: "Author: @utkarsh"
 draft: true
+hide_table_of_contents: true
 keywords:
   - leetcode
   - tutorial
   - binary search tree
   - data structure
 ---
 
-Coming Soon
\ No newline at end of file
+<TutorialAuthors names="@utkarsh"/>
+
+## Overview
+
+Before jump to BST. We recap some basics to better understand BST:
+
+A tree is a data structure composed of nodes that has some following characteristics:
+
+1. Each tree has root node at the top (also known as Parent Node).
+2. The root node has zero or more child nodes.
+3. Each child node has zero or more child nodes. This create subtree in the tree. Every node has its own subtree made up of its children etc. This means that every node on its own can be tree.
+
+A valid **binary search tree** (BST) is defined as follows:
+
+1. The left node of the tree is smaller (or equal) to the root node.
+2. The right node of the tree is greater (or equal) to the root node.
+3. All the nodes of left subtree(tree left side of root node) are smaller (or equal) to the root node.
+4. And All the nodes of right subtree (tree right side of the root node) are greater (or equal) to the root node.
+
+<!-- Diagram of BST -->
+
+![image](https://user-images.githubusercontent.com/30936607/195894102-11f3d201-e1ef-4fd7-83de-5e00e6d66eb7.png)
+
+![image](https://user-images.githubusercontent.com/30936607/195677540-23c39b0f-ce06-4c7d-bffa-468b6c1b3f79.png)
+
+Like binary tree. We can also traverse BST in **pre**,**post** and **inorder** traversal.
+But the amazing fact is that **inorder** traversal
+of `BST` will give **sorted data**.
+
+### Inorder traversal in BST
```

Comment:
use LC problem

Reply:
Here for Inorder traversal, you want LC Problem to explain the Inorder traversal of BST. So I need to search for BST in order in LC and that problem explanation you want